clap_derive/src/attrs.rs

```diff
@@ -543,11 +543,40 @@ impl Attrs {
                         quote!(<#ty as ::std::default::Default>::default())
                     };
 
+                    let container_type = Ty::from_syn_ty(ty);
```

Comment:
> I could see an argument for adding a default_values_t attr, but since default_value_t doesn't work to begin with for Vec<impl ValueEnum>, I feel like we might as well fix default_value_t (plus I personally find it more intuitive, if I used the "typed" version for a Vec, I should be able to pass a Vec).

I think I'd like to start with `default_values_t`

My biggest concern is how resilient the information is that we are keying off of as we add additional types and as users do unexpected things with the API.  `default_values_t` removes the guessing.  We can explore merging them, at a future point.

Reply:
makes sense, one question I have with the `default_values_t` approach is should I then add validation that `container_type == Vec` when `default_values_t` is used?

Reply:
See my other comment about a function with an `impl IntoIterator` parameter for validating the type

That does get messy because it still requires parsing the type.  Bleh.  maybe getting the inner type is ok though?   If its a problem, we can always remove the type check